drone/src/proxy/service.rs

```diff
@@ -271,15 +272,17 @@ impl ProxyService {
                 });
 
                 if req.uri().path() == "/_plane_auth" {
-                    if let Some(query) = req.uri().query() {
-                        if let Some(token) = query.strip_prefix("token=") {
-                            return Ok(Response::builder()
-                                .status(StatusCode::FOUND)
-                                .header("Location", "/")
-                                .header("Set-Cookie", format!("{}={}", PLANE_AUTH_COOKIE, token))
-                                .body(Body::empty())?);
-                        }
-                    }
+                    let params: PlaneAuthParams =
+                        serde_html_form::from_str(req.uri().query().unwrap_or_default())?;
+
+                    return Ok(Response::builder()
+                        .status(StatusCode::FOUND)
+                        .header("Location", params.redirect.as_deref().unwrap_or("/"))
```

Comment:
It might make sense to only redirect to the provided URL if it's on the same domain. Endpoints that blindly redirect based on a query param can be used by bad actors to trick users into visiting sites they didn't want to visit.

Reply:
Good point. Locations can be relative, so maybe rather than match the domain we just ensure that it is a relative path?

Reply:
I think a relative path is simplest and easiest to reason about

Reply:
Done.